lambdas/check-reverification-identity/src/main/java/uk/gov/di/ipv/core/checkreverificationidentity/CheckReverificationIdentityHandler.java

```diff
@@ -0,0 +1,186 @@
+            EvcsService evcsService,
+            UserIdentityService userIdentityService,
+            Gpg45ProfileEvaluator gpg45ProfileEvaluator,
+            VotMatcher votMatcher) {
+        this.configService = configService;
+        this.ipvSessionService = ipvSessionService;
+        this.clientSessionService = clientOAuthSessionDetailsService;
+        this.evcsService = evcsService;
+        this.userIdentityService = userIdentityService;
+        this.gpg45ProfileEvaluator = gpg45ProfileEvaluator;
+        this.votMatcher = votMatcher;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public CheckReverificationIdentityHandler() {
+        this(ConfigService.create());
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public CheckReverificationIdentityHandler(ConfigService configService) {
+        this.configService = ConfigService.create();
+        this.ipvSessionService = new IpvSessionService(configService);
+        this.clientSessionService = new ClientOAuthSessionDetailsService(configService);
+        this.evcsService = new EvcsService(configService);
+        this.userIdentityService = new UserIdentityService(configService);
+        this.gpg45ProfileEvaluator = new Gpg45ProfileEvaluator();
+        this.votMatcher = new VotMatcher(userIdentityService, gpg45ProfileEvaluator);
+    }
+
+    @Tracing
+    @Logging(clearState = true)
+    @Override
+    public Map<String, Object> handleRequest(JourneyRequest request, Context context) {
+        LogHelper.attachComponentId(configService);
+        configService.setFeatureSet(RequestHelper.getFeatureSet(request));
+
+        try {
+            var ipvSessionId = getIpvSessionId(request);
+            LogHelper.attachIpvSessionIdToLogs(ipvSessionId);
+            var ipvSession = ipvSessionService.getIpvSession(ipvSessionId);
+            LogHelper.attachClientSessionIdToLogs(ipvSession.getClientOAuthSessionId());
+            var clientOAuthSession =
+                    clientSessionService.getClientOAuthSession(
+                            ipvSession.getClientOAuthSessionId());
+            LogHelper.attachClientIdToLogs(clientOAuthSession.getClientId());
+            LogHelper.attachGovukSigninJourneyIdToLogs(
+                    clientOAuthSession.getGovukSigninJourneyId());
+
+            var vcs =
+                    evcsService.getVerifiableCredentials(
+                            clientOAuthSession.getUserId(),
+                            clientOAuthSession.getEvcsAccessToken(),
+                            CURRENT);
+
+            var reverificationIdentityVot = getReverificationIdentityVot(vcs);
+            if (reverificationIdentityVot.isEmpty()) {
+                ipvSession.setFailureCode(NO_IDENTITY_AVAILABLE);
+                ipvSessionService.updateIpvSession(ipvSession);
+                return NOT_FOUND_RESPONSE;
+            }
+
+            if (GPG45.equals(reverificationIdentityVot.get().vot().getProfileType())) {
```

Comment:
(regardless of what we do below, I think we still need to do something with the operation profiles!)

Reply:
if we do nothing (leave taget VOT as null), what else is there to do?